src/Files.App/ViewModels/ItemViewModel.cs

```diff
@@ -2012,6 +2018,16 @@ async Task HandleChangesOccurredAsync()
 								App.Logger.LogWarning(ex, ex.Message);
 							}
 
+							if (RenameIndex != -1 && OldName != "" && NewName != "")
+							{
+								var itemRemoved = await RemoveFileOrFolderAsync(OldName);
+								var itemAdded = await AddFileOrFolderAsync(NewName, RenameIndex);
```

Comment:
I don't think you need to assign these two variables, you never read them

```suggestion
								await RemoveFileOrFolderAsync(OldName);
								await AddFileOrFolderAsync(NewName, RenameIndex);
```